src/test/java/org/springframework/data/neo4j/core/transaction/BookmarkForTesting.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2011-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.neo4j.core.transaction;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.neo4j.driver.Bookmark;
+
+/**
+ * @author Michael J. Simons
+ */
+record BookmarkForTesting(String value) implements Bookmark {
+
+	BookmarkForTesting {
+		value = Objects.requireNonNull(value);
+	}
+
+	@Override
+	@SuppressWarnings({ "deprecation", "RedundantSuppression" })
```

Comment:
Is the `SuppressWarnings` only for the call stack? (Just curious)

Reply:
IDEA and the Java compiler can't agree on what they warn about.